### PR TITLE
[lldb] Add test for existential types on the REPL

### PR DESCRIPTION
**Pull request by augusto2112**

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/test/Shell/SwiftREPL/ExistentialTypes.test [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class E: Error, A {}
+
+struct F: Error, A {}
+
+enum G: Error, A { 
+  case empty 
+}
+
+let b: A = B()
+// CHECK: b: B = {}
```
Comment (adrian-prantl) [inline lldb/test/Shell/SwiftREPL/ExistentialTypes.test]: By changing this one line to ```
```
// CHECK: {{b}}: B = {}
```

We can make sure not to accidentally match against the echo of the input comments. I think we're generally disabling echo nowadays, but it doesn't hurt to put in an extra safeguard.

**Review by adrian-prantl — APPROVED** [review]

**Comment by augusto2112** [comment]
@swift-ci test

File: lldb/test/Shell/SwiftREPL/ExistentialTypes.test [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+enum D: A {
+  case empty
+}
+
+class E: Error, A {}
+
+struct F: Error, A {}
+
+enum G: Error, A {
```
Comment (kastiglione) [inline lldb/test/Shell/SwiftREPL/ExistentialTypes.test:21]: Is there something special about `Error`?

File: lldb/test/Shell/SwiftREPL/ExistentialTypes.test [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+let d: A = D.empty
+// CHECK: {{d}}: D = empty
+
+let e: A = E()
+// CHECK: {{e}}: E = {}
+
+let f: A = F()
+// CHECK: {{f}}: F = {}
+
+let g: G = G.empty
```
Comment (kastiglione) [inline lldb/test/Shell/SwiftREPL/ExistentialTypes.test]: Should this be?

```suggestion
let g: A = G.empty
```

**Review by kastiglione — APPROVED** [review]

File: lldb/test/Shell/SwiftREPL/ExistentialTypes.test (hunk 2 above)
Comment (adrian-prantl) [reply lldb/test/Shell/SwiftREPL/ExistentialTypes.test:21]: In SwiftASTContext the Error type is special-cased in a couple of places. I think in TypeSystemSwiftTypeRef it's mostly just another protocol.

Comment (adrian-prantl) [reply lldb/test/Shell/SwiftREPL/ExistentialTypes.test:21]: https://github.com/apple/llvm-project/blob/3c96a2f1c42ab1cf57019e903b03fb1eac682114/lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp#L6629

Comment (adrian-prantl) [reply lldb/test/Shell/SwiftREPL/ExistentialTypes.test:21]: https://github.com/apple/llvm-project/blob/3c96a2f1c42ab1cf57019e903b03fb1eac682114/lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp#L5387

Comment (augusto2112) [reply lldb/test/Shell/SwiftREPL/ExistentialTypes.test:21]: When we try to find the dynamic type of some existential, errors are one of the possible existential cases: https://github.com/apple/swift/pull/37673/files#diff-22f5e159ab66c32ef3537a5a7c2617427545535dd1e7899a837cbf9effe69be1R936

File: lldb/test/Shell/SwiftREPL/ExistentialTypes.test (hunk 3 above)
Comment (augusto2112) [reply lldb/test/Shell/SwiftREPL/ExistentialTypes.test]: Yes! Thanks.

**Comment by augusto2112** [comment]
Failure seems unrelated:

```
21:27:08 PASS: Swift(watchsimulator-i386) :: stdlib/UnicodeTrieGenerator.gyb (14624 of 14625)
22:50:28 
Build timed out (after 83 minutes). Marking the build as failed.
22:50:28 Build was aborted
22:50:28 ninja: build stopped: interrupted by user.
22:50:28 /Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 6 leaked semaphore objects to clean up at shutdown
22:50:28   warnings.warn('resource_tracker: There appear to be %d '
22:50:45 Setting status of 67f38cbef40eb8b3434b36397b7c795a777f575a to FAILURE with url https://ci.swift.org/job/apple-llvm-project-pr-macos/3244/ and message: 'Build finished. '
22:50:45 Using context: Swift Test macOS Platform
22:50:45 Finished: FAILURE
```

**Comment by augusto2112** [comment]
@swift-ci test

**Pull request merged**